### PR TITLE
Fix undo resetting brush size

### PR DESCRIPTION
**Pull request by ClFeSc**
Drawing, then changing brush size, then drawing, then undo, would reset the brush size to its previous value.

### CODE REVIEW

**Review by maximilian-franz — APPROVED** [review]

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st [hunk 1]
```diff
@@ -1,5 +1,6 @@
 undo
-restoreColor: aColor isFilling: aBoolean tool: aSymbol
+restoreBrushSize: aNumber color: aColor isFilling: aBoolean tool: aSymbol
```
Comment (DevSchmidtchen) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st]: Why is the method now called `restoreBrushSize`? It seems to do more than that.

Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st]: It restores the brush size, the color, the filling mode and the tool. Before this push, the method was called `restoreColor:isFilling:tool:`, now it is called `restoreBrushSize:color:isFilling:tool:`. What's wrong with that?

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
+	self brushSize: aNumber.
 	self currentColor: aColor.
 	self isFillingGeometricForm: aBoolean.
 	self currentTool: aSymbol
```
Comment (johannes-poetzsch) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st]: this should be a cascade

Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st]: Good point. Done in latest commit.

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st (hunk 1 above)
Comment (DevSchmidtchen) [reply packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st]: Ah, I see, it looked like the method name only wants to change the brush size, but restores everything else as a 'side effect' (still confused by Smalltalk method names 😅)

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
-	| preColor preFillingGeometric preTool |
+	| preBrushSize preColor preFillingGeometric preTool |
+	preBrushSize := self brushSize.
 	preColor := self currentColor.
 	preFillingGeometric := self isFillingGeometricForm.
 	preTool := self currentTool.
 	
 	aBlock value.
-	self restoreColor: preColor isFilling: preFillingGeometric tool: preTool.
+	self restoreBrushSize: preBrushSize color: preColor isFilling: preFillingGeometric tool: preTool.
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st:10]: A cascade could be used here.

**Review by maximilian-franz — CHANGES_REQUESTED** [review]

**Review by DevSchmidtchen — APPROVED** [review]

**Review by johannes-poetzsch — APPROVED** [review]

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st (hunk 3 above)
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st]: A cascade could be used here.

**Review by maximilian-franz — CHANGES_REQUESTED** [review]

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st (hunk 3 above)
Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st]: You mean something like
```smalltalk
self restoreBrushSize: preBrushSize
    color: preColor
    isFilling: preFillingGeometric
    tool: preTool.
```
? I'm not sure this makes it more readable.

Comment (maximilian-franz) [reply packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st]: Also keep in mind the limit for characters per line (I believe it was 80?)

Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st]: Ok, I'll change it.

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st [hunk 4]
```diff
@@ -0,0 +1,7 @@
+undo
+restoreBrushSize: aNumber color: aColor isFilling: aBoolean tool: aSymbol
+	self
+		brushSize: aNumber;
```
Comment (DevSchmidtchen) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st]: What about the eraser size? Why didn't you use our method `setBrushAndEraserSize`?

Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st]: Good point.

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st (hunk 3 above)
Comment (maximilian-franz) [reply packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st]: > 
> 
> You mean something like
> 
> ```smalltalk
> self restoreBrushSize: preBrushSize
>     color: preColor
>     isFilling: preFillingGeometric
>     tool: preTool.
> ```
> 
> ? I'm not sure this makes it more readable.

No I meant something like that
```smalltalk
self
    restoreBrushSize: preBrushSize color: preColor isFilling: preFillingGeometric tool: preTool;
    reloadCurrentPen
```

Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st]: Oh, ok. This looks reasonable.

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st [hunk 5]
```diff
@@ -1,10 +1,14 @@
...
 	preFillingGeometric := self isFillingGeometricForm.
 	preTool := self currentTool.
 	
 	aBlock value.
-	self restoreColor: preColor isFilling: preFillingGeometric tool: preTool.
+	self restoreBrushSize: preBrushSize
+		color: preColor
+		isFilling: preFillingGeometric
+		tool: preTool.
 	self reloadCurrentPen
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st]: Actually, I meant something like that
```suggestion
	self
		restoreBrushSize: preBrushSize
		color: preColor
		isFilling: preFillingGeometric
		tool: preTool;
		reloadCurrentPen
```

**Review by maximilian-franz — CHANGES_REQUESTED** [review]

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st (hunk 5 above)
Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/M2Backend.class/instance/saveSettingsDuring..st]: But this isn't what is style is meant to be, as far as I know. In this style, the difference between the method call to `restoreBrushAndEraserSize:color:isFilling:tool:` and to `reloadCurrentPen` isn't that obvious.

**Review by maximilian-franz — APPROVED** [review]

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st (hunk 4 above)
Comment (ClFeSc) [reply packages/SketchMorph2-Core.package/M2Backend.class/instance/restoreBrushSize.color.isFilling.tool..st]: Done in the latest commits.

**Review by kpostnov — APPROVED** [review]

**Review by jannis-hpi — APPROVED** [review]

**Pull request merged**